flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java

```diff
@@ -0,0 +1,103 @@
+        Row.of("2020-10-10 00:00:16", 4, 4d, 4f, new BigDecimal("4.44"), "Hi", "b")
+    };
+
+    static final Row[] AFTER_DATA = {
+        Row.of("2020-10-10 00:00:40", 10, 3d, 3f, new BigDecimal("4.44"), "Comment#4", "a"),
+        Row.of("2020-10-10 00:00:42", 11, 4d, 4f, new BigDecimal("5.44"), "Comment#5", "d")
+    };
+    static final TableTestProgram TEMPORAL_SORT_PROCTIME =
+            TableTestProgram.of(
+                            "temporal-sort-proctime", "validates temporal sort node with proctime")
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema(
+                                            "a INT",
+                                            "b BIGINT",
+                                            "c STRING",
+                                            "`proctime` as PROCTIME()")
+                                    .producedBeforeRestore(
+                                            Row.of(1, 1L, "Hi"),
+                                            Row.of(2, 2L, "Hello"),
+                                            Row.of(3, 2L, "Hello world"))
+                                    .producedAfterRestore(
+                                            Row.of(4, 1L, "Guten Morgen"),
+                                            Row.of(5, 2L, "Guten Tag"))
+                                    .build())
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema("a INT")
+                                    .consumedBeforeRestore("+I[1]", "+I[2]", "+I[3]")
+                                    .consumedAfterRestore("+I[4]", "+I[5]")
+                                    .build())
+                    .runSql("INSERT INTO sink_t SELECT a from source_t ORDER BY proctime")
+                    .build();
+
+    static final TableTestProgram TEMPORAL_SORT_ROWTIME =
+            TableTestProgram.of(
+                            "temporal-sort-rowtime", "validates temporal sort node with rowtime")
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema(
+                                            "ts STRING",
+                                            "`int` INT",
+                                            "`double` DOUBLE",
+                                            "`float` FLOAT",
+                                            "`bigdec` DECIMAL(10, 2)",
+                                            "`string` STRING",
+                                            "`name` STRING",
+                                            "`rowtime` AS TO_TIMESTAMP(`ts`)",
+                                            "WATERMARK for `rowtime` AS `rowtime` - INTERVAL '1' SECOND")
```

Comment:
do we need all of the columns if we use only 3?

Reply:
We can certainly remove them.  I used one of the common blocks of data from another restore test.  

I think we can head one of three directions:

1. Share common data.
2. Repeat data in its entirety.
3. Repeat data in a projected form.  

There are arguments for each approach.  The existing tests have use approach 1 and I'm using 2 here. 

Which way would you like / recommend?

Reply:
Does this use data from other test? You define the data here.

The more self-contained the test is, the easier it is to read it without clicking through dozens of clasess.

Reply:
https://github.com/apache/flink/pull/23879/files#diff-522875e33622ba739132c4f1f2bf8ecb4140ff0142c337ead614f26ed343d033R31-R47 is more or less the same as https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupWindowAggregateTestPrograms.java#L31-L51

Reply:
Does it need to be? I see no benefit of using such a complex data for `ORDER BY`